PayloadPkg/FirmwareUpdate/FirmwareUpdate.c

```diff
@@ -1523,6 +1523,11 @@ InitFirmwareUpdate (
         // Start firmware update for the component, exclude CSME driver (CSMD)
         //
         if ((UINT32)ImgHdr->UpdateHardwareInstance != FW_UPDATE_COMP_CSME_DRIVER) {
+          // The CMDI commands should be run after the components have completed the updates
+          if ((UINT32)ImgHdr->UpdateHardwareInstance == FW_UPDATE_COMP_CMD_REQUEST && FwPolicy.Fields.StateMachine != FW_UPDATE_SM_PART_AB) {
```

Comment:
No, since in ApplyFwImage(), the case FW_UPDATE_COMP_CMD_REQUEST doesn't have the ResetRequired set, and the CSME payload is not updated and requests for a reboot, so the only reboot for only one CMDI case is the global fwupdate reboot invoked in EndFirmwareUpdate().

Reply:
Noticed one unnecessary reboot in the case when only one redundant update is run and the update is failed in the first pass, then the system will reboot to go into the second pass and do nothing. But so far I will leave it as is since the failed case is not an usual case.